modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Enrich.scala

```diff
@@ -152,31 +168,48 @@ object Enrich {
     BadRow.GenericError(processor, failure, rawPayload)
   }
 
+  def sinkChunk[F[_]: Concurrent: Parallel, B](
+    results: List[Result],
+    sinkOne: Validated[BadRow, EnrichedEvent] => F[Unit],
+    trackLatency: Option[Long] => F[Unit]
+  ): F[Unit] =
+    results.parTraverse_ {
+      case (events, collectorTstamp) =>
+        events.parTraverse_(one => sinkOne(one)) <* trackLatency(collectorTstamp)
+    }
```

Comment:
That's a good point. Unfortunately we can't have such `.flush` function as all we have access to on the underlying producer is [putData](https://github.com/laserdisc-io/fs2-aws/blob/v3.1.1/fs2-aws/src/main/scala/fs2/aws/internal/KinesisProducerClient.scala#L15).

What we can do however is to use `2` as `concurrency.sink` here https://github.com/snowplow/enrich/blob/feature/enrich-kinesis/modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Enrich.scala#L93 so that next chunk starts getting sunk already when the previous one finishes (and is possibly in the 100ms interval). I'm going to try to see the stats.

`fs2-aws` doesn't add much on top of the underlying `KinesisProducer`, we could also discard it and use our own implementation directly (for producing).

Reply:
Interestingly, the best throughput is observed with `sink.concurrency` set to 2 (= number of cores) and without any `prefetch` (adding `prefetch` has no impact).